packages/flutter_tools/lib/src/android/native_assets.dart

```diff
@@ -0,0 +1,219 @@
+      nativeAssetsBuildMode(buildMode);
+
+  globals.logger
+      .printTrace('Building native assets for $targets $buildModeCli.');
+  final List<Asset> nativeAssets = <Asset>[];
+  final Set<Uri> dependencies = <Uri>{};
+  for (final Target target in targets) {
+    final BuildResult result = await buildRunner.build(
+      linkModePreference: LinkModePreference.dynamic,
+      target: target,
+      buildMode: buildModeCli,
+      workingDirectory: projectUri,
+      includeParentEnvironment: true,
+      cCompilerConfig: await buildRunner.ndkCCompilerConfig,
+      targetAndroidNdkApi: targetAndroidNdkApi,
+    );
+    ensureNativeAssetsBuildSucceed(result);
+    nativeAssets.addAll(result.assets);
+    dependencies.addAll(result.dependencies);
+  }
+  ensureNoLinkModeStatic(nativeAssets);
+  globals.logger.printTrace('Building native assets for $targets done.');
+  if (isAddToApp && nativeAssets.isNotEmpty) {
+    throwToolExit(
+        'Native assets are not yet supported in Android add2app.');
+  }
+  final Map<Asset, Asset> assetTargetLocations =
+      _assetTargetLocations(nativeAssets);
+  await _copyNativeAssetsAndroid(buildUri_, assetTargetLocations, fileSystem);
+  final Uri nativeAssetsUri = await writeNativeAssetsYaml(
+      assetTargetLocations.values,
+      yamlParentDirectory ?? buildUri_,
+      fileSystem);
+  return (nativeAssetsUri, dependencies.toList());
+}
+
+Future<void> _copyNativeAssetsAndroid(
+  Uri buildUri,
+  Map<Asset, Asset> assetTargetLocations,
+  FileSystem fileSystem,
+) async {
+  if (assetTargetLocations.isNotEmpty) {
+    globals.logger
+        .printTrace('Copying native assets to ${buildUri.toFilePath()}.');
+    for (final String jniArchDir in _architectureStringsCMakeAndroid.values) {
+      final Uri archUri = buildUri.resolve('jniLibs/lib/$jniArchDir/');
+      await fileSystem.directory(archUri).create(recursive: true);
+    }
+    for (final MapEntry<Asset, Asset> assetMapping
+        in assetTargetLocations.entries) {
+      final Uri source = (assetMapping.key.path as AssetAbsolutePath).uri;
+      final Uri target = (assetMapping.value.path as AssetAbsolutePath).uri;
+      final String jniArchDir = _architectureStringsCMakeAndroid[
+          assetMapping.value.target.architecture]!;
+      final Uri archUri = buildUri.resolve('jniLibs/lib/$jniArchDir/');
+      final Uri targetUri = archUri.resolveUri(target);
+      final String targetFullPath = targetUri.toFilePath();
+      await fileSystem.file(source).copy(targetFullPath);
+    }
+    globals.logger.printTrace('Copying native assets done.');
+  }
+}
+
+/// Extract the [Target] from an [AndroidArch].
+Target _getNativeTarget(AndroidArch androidArch) {
+  switch (androidArch) {
+    case AndroidArch.armeabi_v7a:
+      return Target.androidArm;
+    case AndroidArch.arm64_v8a:
+      return Target.androidArm64;
+    case AndroidArch.x86:
+      return Target.androidIA32;
+    case AndroidArch.x86_64:
```

Comment:
Consider adding a default case that errors and a test that runs for all AndroidArch values. AndroidArch is about to get another entry as part of the work for risc-v

Reply:
OK forcing the static error makes sense. Sorry I should have realized that but swapping between languages here I forgot that dart does this well compared to other languages.